docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -0,0 +1,575 @@
+===============
+MATCH_RECOGNIZE
+===============
+
+Synopsis
+--------
+
+.. code-block:: text
+
+    MATCH_RECOGNIZE (
+      [ PARTITION BY column [, ...] ]
+      [ ORDER BY column [, ...] ]
+      [ MEASURES measure_definition [, ...] ]
+      [ rows_per_match ]
+      [ AFTER MATCH skip_to ]
+      PATTERN ( row_pattern )
+      [ SUBSET subset_definition [, ...] ]
+      DEFINE variable_definition [, ...]
+      )
+
+Description
+-----------
+
+The ``MATCH_RECOGNIZE`` clause is used to detect patterns across rows of a table. Patterns of interest are
+specified using row pattern syntax based on regular expressions. When the pattern is detected, requested
+information, specified as the row pattern measures, is retrieved from rows that match certain components
+of the pattern.
+
+Example
+-------
+
+In the following example, the pattern describes V-shape over the ``totalprice`` column. A match is found
+whenever orders made by a customer first decrease in price, and then increase past the starting point::
+
+    SELECT * FROM orders MATCH_RECOGNIZE(
+         PARTITION BY custkey
+         ORDER BY orderdate
+         MEASURES
+                  A.totalprice AS starting_price,
+                  LAST(B.totalprice) AS bottom_price,
+                  LAST(D.totalprice) AS top_price
+         ONE ROW PER MATCH
+         AFTER MATCH SKIP PAST LAST ROW
+         PATTERN (A B+ C+ D+)
+         SUBSET U = (B, C)
+         DEFINE
+                  B AS totalprice < PREV(totalprice),
+                  C AS totalprice > PREV(totalprice) AND totalprice <= A.totalprice,
+                  D AS totalprice > PREV(totalprice)
+         )
+
+Below, all subclauses of ``MATCH_RECOGNIZE`` clause are explained, using this example as illustration.
+
+Partitioning and ordering
+-------------------------
+
+The ``orders`` table, which is the input table of row pattern recognition, is partitioned by the column ``custkey``::
+
+    PARTITION BY custkey
+
+When partitioning is specified, pattern matching is performed independently on every partition. Otherwise,
```

Comment:
The PARTITION BY statement allows you to break up the input table into separate sections, that are independently processed for pattern matching. Without a partition declaration, the whole input table is used. This behavior is is analogous to the semantics in the WINDOW clause. In the example, the orders table is partitioned by the custkey value, so that pattern matching is performed for all orders of a specific customers independently from orders of other customers.

Reply:
Above needs syntax and links..